qiskit/algorithms/eigen_solvers/vqd.py

```diff
@@ -0,0 +1,840 @@
+        Optional[BOUNDS],  # parameters bounds for the optimization
+    ],
+    RESULT,  # a result object (either SciPy's or Qiskit's)
+]
+
+# TODO: Add a more detailed explanation of the algorithm in the main docstring
+
+
+class VQD(VariationalAlgorithm, Eigensolver):
+    r"""The Variational Quantum Deflation algorithm.
+
+    `VQD <https://arxiv.org/abs/1805.08138>`__ is a quantum algorithm that uses a
+    variational technique to find
+    the k eigenvalues of the Hamiltonian :math:`H` of a given system.
+
+    An instance of VQD requires defining three algorithmic sub-components:
+    an integer k denoting the number of eigenstates to calculate, a trial
+    state (a.k.a. ansatz)which is a :class:`QuantumCircuit`,
+    and one of the classical :mod:`~qiskit.algorithms.optimizers`.
+    The ansatz is varied, via its set of parameters, by the optimizer,
+    such that it works towards a state, as determined by the parameters
+    applied to the ansatz, that will result in the minimum expectation values
+    being measured of the input operator (Hamiltonian). The algorithm does
+    this by iteratively refining each excited state to be orthogonal to all
+    the previous excited states.
+
+    An optional array of parameter values, via the *initial_point*, may be provided as the
+    starting point for the search of the minimum eigenvalue. This feature is particularly useful
+    such as when there are reasons to believe that the solution point is close to a particular
+    point.
+
+    The length of the *initial_point* list value must match the number of the parameters
+    expected by the ansatz being used. If the *initial_point* is left at the default
+    of ``None``, then VQE will look to the ansatz for a preferred value, based on its
+    given initial state. If the ansatz returns ``None``,
+    then a random point will be generated within the parameter bounds set, as per above.
+    If the ansatz provides ``None`` as the lower bound, then VQE
+    will default it to :math:`-2\pi`; similarly, if the ansatz returns ``None``
+    as the upper bound, the default value will be :math:`2\pi`.
+
+    """
+
+    def __init__(
+        self,
+        ansatz: Optional[QuantumCircuit] = None,
+        k: int = 2,
+        betas: Optional[List[float]] = None,
+        optimizer: Optional[Union[Optimizer, MINIMIZER]] = None,
+        initial_point: Optional[np.ndarray] = None,
+        gradient: Optional[Union[GradientBase, Callable]] = None,
+        expectation: Optional[ExpectationBase] = None,
+        include_custom: bool = False,
+        max_evals_grouped: int = 1,
+        callback: Optional[Callable[[int, np.ndarray, float, float], None]] = None,
+        quantum_instance: Optional[Union[QuantumInstance, Backend]] = None,
+    ) -> None:
+        """
+
+        Args:
+            ansatz: A parameterized circuit used as ansatz for the wave function.
+            optimizer: A classical optimizer.
```

Comment:
Can you order these in the same order as constructor. Maybe you can check what VQE has for equivalent arguments, like optimizer, and copy the text it has. While this may have started from the same point I think VQE has altered - at least it has far more text describing the optimizer parameter.